code/datums/status_effects/debuffs.dm

```diff
@@ -12,12 +13,12 @@
 	if(.)
 		if(updating_canmove)
 			owner.update_canmove()
-			if(issilicon(owner))
+			if(needs_update_stat || issilicon(owner))
 				owner.update_stat()
 
 /datum/status_effect/incapacitating/on_remove()
 	owner.update_canmove()
-	if(issilicon(owner)) //silicons need stat updates in addition to normal canmove updates
+	if(needs_update_stat || issilicon(owner)) //silicons need stat updates in addition to normal canmove updates
```

Comment:
you could have removed ```issilicon()``` by setting ```needs_update_stat``` if the owner was a silicon on effect creation...